src/Uno.UI/Microsoft/UI/Xaml/Controls/InfoBar/InfoBar.Events.cs

```diff
@@ -0,0 +1,13 @@
+﻿using Windows.Foundation;
+
+namespace Microsoft.UI.Xaml.Controls
+{
+	public partial class InfoBar
```

Comment:
Is this what we are doing now? Actually, I always like splitting out Events and Properties in separate files as that matches the documentation structure and is also latest convention with c# controls. However, this is not done in WinUI and I thought fidelity with original code was highest priority?

We should be fairly consistent with ported controls and add .Events.cs for the others if that's what we agree.

Reply:
I have been doing this for the controls I have ported so far. I think it is not a big problem as the events are defined in the header file/.idl anyway, so if we really wanted to be precise, we would have to split the structure to "cpp" and "h" or something like that. The WinUI porting story will definitely evolve over time, so I would just opt for an approach that works and we can refactor as needed over time. The public interface will remain the same anyway :-) .

Reply:
Actually, the events are also in the generated [.properties.cpp files](https://github.com/microsoft/microsoft-ui-xaml/blob/f5034668462987302f1e1c58d1e055e45e40c658/dev/Generated/InfoBar.properties.cpp#L437). Obviously we merge .cpp and .h files together into one .cs file for c#. The IDL is only used to generate the source code and set initial values. We shouldnt use it here other than to make sure default values match. In the future we could generate properties based on the IDL just like WinUI though. I'm getting off track, my point was everything is in .properties.cs in WinUI.

Reply:
I see, I didn't notice that. I can move the events in Properties file, but need to adjust my previous controls the same way too

Reply:
Well, I wouldn't redo anything in past controls. It can be changed when the control is updated to latest WinUI next. Or forgotten about since it works just fine :) Probably we should keep everything in the properties file for new contrils. Maybe change that when/if we generate from IDL. I will file an issue for some ideas today as to keep in sync long term we will need some automation.